Test/PublishNodeModule.Tests.ps1

```diff
@@ -157,115 +150,202 @@ function ThenTaskFailed
-    try
-    {
-        Init
-        New-PublishNodeModuleStructure
-        WhenPublishingNodeModule
-        ThenNpmrcCreated
-        ThenNpmPackagesPruned
-        ThenNodeModulePublished
-    }
-    finally
-    {
-        Remove-Node
+    It 'should publish the module' {
+        try
+        {
+            Init
+            GivenPackageJson @"
+            {
+                "name": "publishnodemodule_test",
+                "version": "1.2.0"
+            }
+"@
+            WhenPublishingNodeModule -WithCredentialID 'NpmCred' `
+                                     -WithEmailAddress 'somebody@example.com' `
+                                     -WithNpmRegistryUri 'http://registry@example.com'
+            ThenNpmrcCreated -WithEmail 'somebody@example.com' -WithRegistry 'http://registry@example.com'
+            ThenNpmPackagesPruned
+            ThenNodeModulePublished
+        }
+        finally
+        {
+            Remove-Node
+        }
     }
 }
 
 Describe 'PublishNodeModule.when NPM registry URI property is missing' {
-    try
-    {
-        Init
-        GivenNoNpmRegistryUri
-        New-PublishNodeModuleStructure
-        WhenPublishingNodeModule -ErrorAction SilentlyContinue
-        ThenTaskFailed '\bNpmRegistryUri\b.*\bmandatory\b'
-    }
-    finally
-    {
-        Remove-Node
+    It 'should fail' {
+        try
+        {
+            Init
+            GivenPackageJson @"
+            {
+                "name": "publishnodemodule_test",
+                "version": "1.2.0"
+            }
+"@
+            WhenPublishingNodeModule -WithCredentialID 'NpmCred' `
+                                     -WithEmailAddress 'somebody@example.com' `
+                                     -ErrorAction SilentlyContinue
+            ThenTaskFailed '\bNpmRegistryUri\b.*\bmandatory\b'
+        }
+        finally
+        {
+            Remove-Node
+        }
     }
 }
 
 Describe 'PublishNodeModule.when credential ID property missing' {
-    try
-    {
-        Init
-        GivenNoCredentialID
-        New-PublishNodeModuleStructure
-        WhenPublishingNodeModule -ErrorAction SilentlyContinue
-        ThenTaskFailed '\bCredentialID\b.*\bmandatory\b'
-    }
-    finally
-    {
-        Remove-Node
+    It 'should fail' {
+        try
+        {
+            Init
+            GivenPackageJson @"
+            {
+                "name": "publishnodemodule_test",
+                "version": "1.2.0"
+            }
+"@
+            WhenPublishingNodeModule -WithEmailAddress 'somebody@example.com' `
+                                     -WithNpmRegistryUri 'http://registry@example.com' `
+                                     -ErrorAction SilentlyContinue
+            ThenTaskFailed '\bCredentialID\b.*\bmandatory\b'
+        }
+        finally
+        {
+            Remove-Node
+        }
     }
 }
 
 Describe 'PublishNodeModule.when email address property missing' {
+    It 'should fail' {

```

Comment:
Now that we're using a single `It` per `Describe`, we can use Pester's `AfterEach` to handle the Node removal. Create a `Reset` function and remove all the `try`/`catch` blocks and add an `AfterEach { Reset }` to each `Describe`:

```powershell
function Reset
{
    Remove-Node
}

Describe '' {
    AfterEach { Reset }
    It '' {
    }
}
```

Reply:
I was going to do this but then decided to leave it inside the `It` block so that the standup and teardown tasks are included in the `It` block timings, which I believe is the point of the single It-per-Describe pattern?

Reply:
The single `It` is to get more accurate timings. I didn't know if Pester includes before/after blocks in its timings, so I did a little test:

```powershell
Describe 'Timings' {
    BeforeEach { Start-Sleep 1 }
    AfterEach { Start-Sleep 1 }
    It 'should time' {
        $true | Should -Be $true
    }
}


Describing Timings
  [+] should time 2.16s
```

Looks like it does. Woot! (I'm glad, otherwise I'd have to find an argument for why I like the `AfterEach` over `try`/`finally` that isn't "I just think it looks better.") 😆

Reply:
Ah great! \( ﾟヮﾟ)/